app/helpers/money_helper.rb

```diff
@@ -1,11 +1,17 @@
 module MoneyHelper
   include MoneyRails::ActionViewExtension
 
+  PERMITTED_CURRENCIES = %w[USD EUR CAD].freeze
+
   # TODO: Add a bank implementation that fetches conversion rate values
-  Money.default_bank.add_rate(:USD, :EUR, 0.88)
+  Money.default_bank.add_rate(:USD, :EUR, 0.89)
+  Money.default_bank.add_rate(:USD, :CAD, 1.32)
 
   def default_currency
-    t(I18n.locale, scope: [:money, :currencies])
+    currency = t(I18n.locale, scope: [:money, :currencies])
```

Comment:
Aren't we only setting `I18n.locale` to permitted locales?

Reply:
Should this be request locale? Or something?

I don’t understand why this is happening, frankly.

Reply:
> Aren't we only setting I18n.locale to permitted locales?

Oh we're actually not -- that'd be why

```rb
# app/controllers/application_controller.rb L66-72 

def requested_locale
  @requested_locale ||=
    locale_from_request_params ||
    current_user&.preferred_language.presence ||
    locale_from_request_header ||
    I18n.default_locale
end
```

Reply:
Hastily put together, but something like this might resolve the issue upstream?

```rb
# app/controllers/application_controller.rb L66-78 

def requested_locale
  return @requested_locale if defined?(@requested_locale)

  @requested_locale ||=
    locale_from_request_params ||
    current_user&.preferred_language.presence ||
    locale_from_request_header ||
    I18n.default_locale

  unless @requested_locale.in?(I18n.available_locales)
    @requested_locale = I18n.default_locale
  end
end
```

Reply:
I think we need to create a test that replicates the actual issue and then resolve it without rescuing random things. I'll get to that later this afternoon

Reply:
Reproduced and tested manually via browser settings -- we have some requests specs that send an invalid locale, I'm not sure why those aren't triggering the exception though.

Reply:
Me either!

Want to take this over @jmromer ? I don't think we need to investigate more deeply, if you can set headers in the locale request and trigger the error, that's good enough.

Reply:
No prob -- I think the see the problem in the specs